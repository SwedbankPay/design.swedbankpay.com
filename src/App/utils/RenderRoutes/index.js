import React from "react";
import PropTypes from "prop-types";
import { Switch, Route, Redirect } from "react-router-dom";

import { DocHeading, StatusBadge } from "@docutils";

const RenderRoutes = ({ path, redirect, routes, appFolder }) => (
    <Switch>
        <Route exact path={path} render={() => <Redirect to={redirect} />}/>
        {routes.map(route => {
            const { path, componentPath } = route;

            const RouteRenderComponent = React.lazy(() => import(/* webpackChunkName: "doc-route.chunk_" */ `../../${appFolder}/${componentPath}/index.js`));

            return <Route key={`doc_route_${path}`} exact path={path} render={() => <>
                <div className="d-flex align-items-center ">
                    <DocHeading />
                    {route.statusBadges && route.statusBadges.map((statusBadge, i) => <StatusBadge key={i} type={statusBadge} />)}
                </div>
                <RouteRenderComponent />
            </>} />;
        })}
        <Redirect from={`${path}/*`} to="/404" />
    </Switch>
);

RenderRoutes.propTypes = {
    path: PropTypes.string.isRequired,
    redirect: PropTypes.string.isRequired,
    routes: PropTypes.array.isRequired,
    appFolder: PropTypes.string.isRequired
};

export default RenderRoutes;
