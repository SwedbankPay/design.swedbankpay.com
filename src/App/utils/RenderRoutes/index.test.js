import RenderRoutes from "./index";

describe("Utilities: RenderRoutes", () => {
	it("is defined", () => {
		expect(RenderRoutes).toBeDefined();
	});
});
