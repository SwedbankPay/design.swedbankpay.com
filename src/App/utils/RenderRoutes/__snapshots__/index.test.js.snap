// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: RenderRoutes renders a basic route structure 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "testKey",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "createURL": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "index": 0,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <renderRoutes
    appFolder="test"
    path="test"
    redirect="test"
    routes={
      Array [
        Object {
          "component": Object {
            "default": "test-2",
          },
          "path": "test-2",
        },
      ]
    }
  />
</Router>
`;
